packages/tests/tests/peer_exchange.node.spec.ts

```diff
@@ -43,45 +42,9 @@ describe("Peer Exchange", () => {
     // we want to ensure that there is enough time for discv5 to discover peers
     await delay(40000);
 
-    await waitForRemotePeer(waku, [Protocols.PeerExchange]);
-    const pxPeers = await waku.peerExchange.peers();
-    expect(pxPeers.length).to.be.greaterThan(0);
-  });
-
-  it("Manual query on test fleet", async function () {
-    this.timeout(60_000);
-
-    const waku = await createLightNode({
-      libp2p: {
-        peerDiscovery: [
-          bootstrap({ list: getPredefinedBootstrapNodes(Fleet.Test) }),
-        ],
-      },
-    });
-
-    await waku.start();
-
-    await waitForRemotePeer(waku, [Protocols.PeerExchange]);
-
-    let receivedCallback = false;
-    const numPeersToRequest = 3;
-    const callback = (response: PeerExchangeResponse): void => {
-      receivedCallback = true;
-      expect(response.peerInfos.length).to.be.greaterThan(0);
-      expect(response.peerInfos.length).to.be.lessThanOrEqual(
-        numPeersToRequest
-      );
-
-      expect(response.peerInfos[0].ENR).to.not.be.null;
-    };
-
-    await waku.peerExchange.query(
-      {
-        numPeers: numPeersToRequest,
-      },
-      callback
-    );
-
-    expect(receivedCallback).to.be.true;
+    const pxPeers = waku.libp2p
+      .getConnections()
```

Comment:
This strategies makes the test slow to run every time.

Instead of waiting 40s and then checking connection do this: listen on new connection events on `libp2p`. for each new connection, check tag. As soon as one with a peer exchange tag is found, test can stop.

You can also wrap that in a timeout so it does not wait forever.